client/components/decks.jsx

```diff
@@ -1,21 +1,77 @@
 import React from 'react';
+import Modal from './modal';
+import NewDeck from './newdeck';
+import DeleteDeck from './deletedeck';
 
 export default class Decks extends React.Component {
   constructor(props) {
     super(props);
-    this.state = ({ currentShowing: null, lastShowing: null });
+    this.state = ({
+      currentShowing: null,
+      lastShowing: null,
+      show: false,
+      form: null,
+      deleteDeckId: null
+    });
     this.showOptions = this.showOptions.bind(this);
     this.hideOptions = this.hideOptions.bind(this);
+    this.showModal = this.showModal.bind(this);
+    this.closeModal = this.closeModal.bind(this);
   }
 
   showOptions(event) {
     const lastShowing = this.state.currentShowing;
-    this.setState({ currentShowing: event.target.id, lastShowing });
+    this.setState({
+      currentShowing: event.target.id,
+      lastShowing
+    });
   }
 
   hideOptions(event) {
     const lastShowing = this.state.currentShowing;
-    this.setState({ currentShowing: null, lastShowing });
+    this.setState({
+      currentShowing: null,
+      lastShowing
+    });
+  }
+
+  showModal(event) {
+    const form = event.target.id;
+    if (form === 'deletedeck') {
+      const deleteDeckId = (event.target.closest('.scene').getAttribute('id'));
+      this.setState({
+        show: true,
+        form: event.target.id,
+        deleteDeckId
+      });
+    } else {
+      this.setState({
+        show: true,
+        form
+      });
+    }
+
+  }
+
+  closeModal(event) {
+    this.props.updateDecks();
+    this.setState({
+      show: false,
+      form: null,
+      deleteDeckId: null
+    });
+  }
+
+  renderModalForm() {
+    const { form } = this.state;
+    if (form === 'newdeck') {
+      return <NewDeck closeModal={this.closeModal} />;
+    } else if (form === 'deletedeck') {
+      return <DeleteDeck
+        deckId={this.state.deleteDeckId}
+        closeModal={this.closeModal}
+      />;
```

Comment:
Since we already have all of the decks, can we just pass the targeted deck as a prop to the delete modal? That way we don't need the extra fetch in there

Reply:
That makes sense to make it quicker and also just remove the extra request. Here I added an id to the containing section for the button with the showModal function and used that id for the deck name top ass as a prop. I also specified a closeModal vs submitModal to not send a fetch request is a use just wants to close the modal (since this would update 0 decks)

Targetting the deckname in decks.jsx

![image](https://user-images.githubusercontent.com/109567499/201726708-f4fe1fd4-1ad9-467b-b974-02e71dc21f57.png)

Passing as prop to delete deck modal

![image](https://user-images.githubusercontent.com/109567499/201726776-ae8e37de-d9ec-43de-bda3-02f1df3be621.png)

Using the props instead of state for the deck name

![image](https://user-images.githubusercontent.com/109567499/201726901-1426f2d7-0836-4545-8833-02e1ac1af219.png)

![modal update workinng](https://user-images.githubusercontent.com/109567499/201727411-5e180b0d-79b8-487a-9202-9bc4aab1d435.gif)